src/crontab.ts

```diff
@@ -0,0 +1,396 @@
+    } else {
+      throw new Error(
+        `Options on line ${lineNumber} of crontab contains unsupported key '${key}'; supported keys are: '${Object.keys(
+          matchers,
+        ).join("', '")}'.`,
+      );
+    }
+  });
+
+  if (!backfillPeriod) {
+    backfillPeriod = 0;
+  }
+
+  return {
+    options: { backfillPeriod, maxAttempts, queueName, priority },
+    identifier,
+  };
+};
+
+const parseCrontabPayload = (
+  lineNumber: number,
+  payloadString: string | undefined,
+): any => {
+  if (!payloadString) {
+    return null;
+  }
+  try {
+    return JSON5.parse(payloadString);
+  } catch (e) {
+    throw new Error(
+      `Failed to parse JSON5 payload on line ${lineNumber} of crontab: ${e.message}`,
+    );
+  }
+};
+
+const parseCrontabCommand = (
+  lineNumber: number,
+  command: string,
+): Pick<CronItem, "task" | "options" | "payload" | "identifier"> => {
+  const matches = CRONTAB_COMMAND.exec(command);
+  if (!matches) {
+    throw new Error(
+      `Invalid command specification in line ${lineNumber} of crontab.`,
+    );
+  }
+  const [, task, optionsString, payloadString] = matches;
+  const { options, identifier = task } = parseCrontabOptions(
+    lineNumber,
+    optionsString,
+  );
+  const payload = parseCrontabPayload(lineNumber, payloadString);
+  return { task, options, payload, identifier };
+};
+
+/**
+ * Parses a line from a crontab file, such as `* * * * * my_task`
+ */
+export const parseCrontabLine = (
+  crontabLine: string,
+  lineNumber: number,
+): CronItem => {
+  const matches = CRONTAB_LINE_PARTS.exec(crontabLine);
+  if (!matches) {
+    throw new Error(
+      `Could not process line '${lineNumber}' of crontab: '${crontabLine}'`,
+    );
+  }
+  const minutes = parseCrontabRange(
+    `range 1 in line ${lineNumber} of crontab`,
+    matches[1],
+    0,
+    59,
+  );
+  const hours = parseCrontabRange(
+    `range 2 in line ${lineNumber} of crontab`,
+    matches[2],
+    0,
+    23,
+  );
+  const dates = parseCrontabRange(
+    `range 3 in line ${lineNumber} of crontab`,
+    matches[3],
+    1,
+    31,
+  );
+  const months = parseCrontabRange(
+    `range 4 in line ${lineNumber} of crontab`,
+    matches[4],
+    1,
+    12,
+  );
+  const dows = parseCrontabRange(
+    `range 5 in line ${lineNumber} of crontab`,
+    matches[5],
+    0,
+    6,
+    true,
+  );
+  const { task, options, payload, identifier } = parseCrontabCommand(
+    lineNumber,
+    matches[6],
+  );
+
+  return {
+    minutes,
+    hours,
+    dates,
+    months,
+    dows,
+    task,
+    options,
+    payload,
+    identifier,
+  };
+};
+
+export const parseCrontab = (crontab: string): Array<CronItem> => {
+  const lines = crontab.split(/\r?\n/);
+  const items: CronItem[] = [];
+  for (
+    let lineNumber = 1, numberOfLines = lines.length;
+    lineNumber <= numberOfLines;
+    lineNumber++
+  ) {
+    const line = lines[lineNumber - 1];
+    if (line.startsWith("#") || line.trim() === "") {
```

Comment:
Need to think more about comments, particularly where `#` may occur in the rest of the line.

Reply:
> Blank lines and leading spaces and tabs are ignored. Lines whose first non-space character is a pound-sign (#) are comments, and are ignored. Note that comments are not allowed on the same line as cron commands, since they will be taken to be part of the command. Similarly, comments are not allowed on the same line as environment variable settings.

Cannot do inline comments in cron :+1: